### PR TITLE
Aquarium QoL: Fish bag, lower hunger rate and more

### LINKED ISSUE #10848: AI can tap Fishtanks
**Problem Description**:
The AI is able to perform the tap action by clicking on a fishtank

**What did you expect to happen**: 
The AI should not be able to interact with a physical object through their 'Eye'

**What happened instead**:
It made the 'tapping' chat message and sound

**Why is this bad/What are the consequences:**
This is a minor issue, as I can't think of any way to exploit it, but it is funny.

**Steps to reproduce the problem**:
Be the AI. Have someone set up a fishtank, unsure if actual fish needed. Click on the fishtank when observing the room as the AI.

**When did the problem start happening**:
Unknown

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was anything else out of the ordinary happening?)**:
Don't know if it's a minor fix, but it's not a major issue.

### PR DESCRIPTION
**Pull request by dearmochi**
## What Does This PR Do

- **Adds the bag Fish Bag. It comes in the aquatic starter kit and can also be crafted from cloth. It can be used to quickly pick up fish and fish eggs, similar to plant bags.**
  - Scooping up fish or fish eggs while holding a fish bag will move the items directly into the bag rather than at the user's feet.
- **Add a fish overlay on wall aquariums that varies depending on the number of fish inside.**
- **Reduce the food depletion rate in aquariums by half.**
- Scooping up fishes will now show the fish count per type in the input list, making it easier to track which fish types are duplicate.
- Scooping up eggs will no longer yield dud fish eggs to prevent cluttering from functionally useless items. A message saying how many duds there were will appear instead.
- Fixes #10848

## Why It's Good For The Game

- The fish bag and the associated changes with scooping fishes/fish eggs make it less of a chore to manage aquariums and the harvested items, especially since they don't drop to the ground if the user holds a fish bag.
- The food depletion rate being lowered is a good thing because fish would get hungry **very frequently**. Halving the rate ensures there is less micromanagement involved.
- The visual change is there for immersion, but also helps in telling at a glance how many fish there are in an aquarium.
- Dud eggs not being obtainable anymore is a good thing because they literally did nothing but serve as clutter.

## Images of changes

bag
_Fish bag_

  
_The fish overlay that shows depending on the number of fish inside_

## Changelog
:cl:
add: Add the Fish Bag which allows for easy collection of fish and fish eggs from aquariums
add: Holding a fish bag while harvesting fish and fish eggs will move the items inside the bag rather than the floor
add: Add a fish overlay on wall aquariums that changes depending on the number of fish inside
add: Display the number of fish per type in the input when harvesting fish
tweak: Aquarium fish now get hungry 50% slower
tweak: Dud fish eggs can no longer be harvested
fix: Fix AIs being able to tap on fish tanks
/:cl:

### CODE REVIEW

**Review by AffectedArc07 — APPROVED** [review]
Code seems solid enough

**Review by SteelSlayer — APPROVED** [review]

**Comment by Mitchs98** [comment]
I love you.

File: code/modules/fish/fishtank.dm [hunk 1]
```diff
@@ -279,44 +294,68 @@
...
 
 	if(egg_count > max_fish)						//Make sure the number of eggs doesn't exceed the max_fish for the tank
 		egg_count = max_fish						//If you somehow exceeded the cap, set the egg_count to max, destroy the excess later
 
+	var/duds = 0
 	while(egg_count > 0)							//Loop until you've harvested all the eggs
 		var/obj/item/fish_eggs/egg = pick(egg_list)	//Select an egg at random
-		egg = new egg(get_turf(user))				//Spawn the egg at the user's feet
+		if(egg != /obj/item/fish_eggs) 				// Don't harvest duds
+			egg = new egg(fish_bag || get_turf(user))			//Spawn the egg at the user's feet
```
Comment (dearmochi) [inline code/modules/fish/fishtank.dm:313]: This doesn't respect the bag's capacity

File: code/modules/fish/fishtank.dm [hunk 2]
```diff
@@ -279,44 +294,68 @@
...
+			user.visible_message("[user.name] scoops \a [fish_name] from [src] into [fish_bag].", "You scoop \a [fish_name] out of [src] into [fish_bag].")
+		else
+			user.visible_message("[user.name] scoops \a [fish_name] from [src].", "You scoop \a [fish_name] out of [src].")
+		// Move the fish in
+		var/fish_item = fish_to_scoop.fish_item
 		if(fish_item)
-			new fish_item(get_turf(user))			//Spawn the appropriate fish_item at the user's feet.
-		kill_fish(fish_type)						//Kill the caught fish from the tank
-													//Update the air for the turf, to avoid permanent atmos sealing with wall tanks
+			new fish_item(fish_bag || get_turf(user))	//Spawn the appropriate fish_item at the user's feet or in their held fish bag.
```
Comment (dearmochi) [inline code/modules/fish/fishtank.dm:356]: Ditto

File: code/modules/fish/fishtank.dm [hunk 3]
```diff
@@ -346,14 +348,15 @@
...
-			user.visible_message("[user.name] scoops \a [fish_name] from [src].", "You scoop \a [fish_name] out of [src].")
 		// Move the fish in
 		var/fish_item = fish_to_scoop.fish_item
 		if(fish_item)
-			new fish_item(fish_bag || get_turf(user))	//Spawn the appropriate fish_item at the user's feet or in their held fish bag.
+			var/obj/item/I = new fish_item(get_turf(user))
+			if(fish_bag?.can_be_inserted(I))
+				fish_bag.handle_item_insertion(I)
+			else
+				fish_bag = null
```
Comment (SteelSlayer) [inline code/modules/fish/fishtank.dm:358]: Do you need to set this to null here? The variable isn't used again after this point.

Comment (dearmochi) [reply code/modules/fish/fishtank.dm:358]: Whoops

**Comment by Carthusia** [comment]
Yeeeeeeesssssssssssss, death to dud fish eggs, all hail the fish egg bag!

**Comment by PacifistDalek** [comment]
Between this and the conveyor change, its sushi time baybeee

**Review by variableundefined — APPROVED** [review]
Approved for design

**Review by Kyep — APPROVED** [review]

**Pull request merged**